### DISCUSSION TITLE
Do i need to specify the built in plugins if i start using other plugins?

### DISCUSSION THREAD

**Discussion by e4rthdog** [squidfunk/mkdocs-material#6934] [Q&A] [answered]
My` mkdocs.yml` file did not have an entry for plugin definition and everything worked (search e.t.c).

After i added a plugin: line in the file for another third party plugin `mkdocs-nav-weight` then i lost search.

I found out that i need to add search in the `plugins:` section in order to get it back again.

Do i need to specify all built in plugins also?

**Comment by alexvoss** [comment, marked as answer]
The search plugin is the only one that is turned on by default, so only it needs to be explicitly mentioned in the configuration when you create a `plugins` entry. The other ones need to be configured explicitly anyway.

**Reply by rrakso** [reply]
@alexvoss Is it possible to use plugins other than the built-in ones? I wanted to use [mkdocs-print-site-plugin
](https://pypi.org/project/mkdocs-print-site-plugin/).

**Reply by kamilkrzyskow** [reply]
@rrakso
Yes, you can use external plugins to a major extent. Some plugins might interfere with the HTML output and therefore might break styling in the Material theme, but considering that squidfunk chipped in as the contributor to this plugin, then everything should work out ;)

Follow the installation instructions and if it's your first installed plugin remember to keep the `search` entry in the `plugins` section:
```yml
plugins:
  - search
  - print-site
```

**Reply by alexvoss** [reply]
Yep, what @kamilkrzyskow said. It is somewhat odd that the search plugin is turned on by default and then needs to be added explicitly once other plugins are listed. IMHO a better approach would be to just generate a plugins section in `mkdocs new` that contains search as the only one. Not top of my list of hills to fight and die on, though.

**Reply by rrakso** [reply]
Thank you so much guys! I would like to point out that I searched the Internet for information on how I can use add-ons together with this docker image, unfortunately I did not find.... Only by reading your comments I got the idea to look for information in the official documentation and guess what? Of course I found it hahaha -> [How to add plugins to the Docker image?
](https://squidfunk.github.io/mkdocs-material/getting-started/#:~:text=How%20to%20add%20plugins%20to%20the%20Docker%20image%3F)

**Reply by squidfunk** [reply]
> Some plugins might interfere with the HTML output and therefore might break styling in the Material theme, but considering that squidfunk chipped in as the contributor to this plugin, then everything should work out ;)

FYI, I'm not an active contributor. @timvink added me as a backup, as I understand, but I haven‘t yet contributed anything 😅

**Reply by timvink** [reply]
Haha yeah @squidfunk was kind enough to be backup for my plugins. Should I ever disappear off the radar he could always appoint new maintainers.
